### PR TITLE
Fix upgrade documentation about form handling

### PR DESCRIPTION
**Pull request by JulienMattiussi**
- [x] fix useCreate on upgrade doc

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -161,7 +161,6 @@ const SaveWithNoteButton = props => {
         }
 
         create(
-            null,
             {
                 data: { ...formState.values, average_note: 10 },
```
Comment (djhi) [inline UPGRADE.md]: Shouldn't it be the following ?

```suggestion
                payload: { data: { ...formState.values, average_note: 10 }, },
```

Comment (JulienMattiussi) [reply UPGRADE.md]: right

File: UPGRADE.md [hunk 2]
```diff
@@ -374,6 +375,36 @@ export default ({
...
+
+[Final form migration documentation here](https://final-form.org/docs/react-final-form/migration/redux-form) explains the various changes you have to perform in your code.
+
+```diff
+-import { reduxForm } from 'redux-form'
++import { Form } from 'react-final-form'
+
+-const CustomForm = reduxForm({ form: 'record-form', someOptions: true })(({ record, resource }) => (	
++const CustomForm = ({ record, resource }) => (
++   <Form someOptions={true}>
```
Comment (djhi) [inline UPGRADE.md]:
```suggestion
+    <Form someOptions={true}>
+        {({ handleSubmit }) => (
```

File: UPGRADE.md [hunk 3]
```diff
@@ -374,6 +375,36 @@ export default ({
...
+    <div>
```
Comment (djhi) [inline UPGRADE.md]:
```suggestion
    +            <form onSubmit={handleSubmit}>
```

File: UPGRADE.md [hunk 4]
```diff
@@ -374,6 +375,36 @@ export default ({
...
+-import { reduxForm } from 'redux-form'
++import { Form } from 'react-final-form'
+
+-const CustomForm = reduxForm({ form: 'record-form', someOptions: true })(({ record, resource }) => (	
++const CustomForm = ({ record, resource }) => (
++   <Form someOptions={true}>
+    <div>	    
+        <Typography>Notes</Typography>
+        <TextInput source="note" />
+    </div>
```
Comment (djhi) [inline UPGRADE.md]:
```suggestion
    </form>
```

File: UPGRADE.md [hunk 5]
```diff
@@ -374,6 +375,36 @@ export default ({
...
++   </Form>
```
Comment (djhi) [inline UPGRADE.md]:
```suggestion
+        )}
+    </Form>
```

Comment (djhi) [reply UPGRADE.md]: The whole snippet might need a little reformatting

File: UPGRADE.md [hunk 6]
```diff
@@ -374,6 +375,40 @@ export default ({
 };
 ```
 
+## Custom Forms using `reduxForm()` must be replaced by using `<Form>`
+
+[Final form migration documentation here](https://final-form.org/docs/react-final-form/migration/redux-form) explains the various changes you have to perform in your code.
```
Comment (fzaninotto) [inline UPGRADE.md]:
```suggestion
The [final-form migration documentation here](https://final-form.org/docs/react-final-form/migration/redux-form) explains the various changes you have to perform in your code.
```

File: UPGRADE.md [hunk 7]
```diff
@@ -374,6 +375,40 @@ export default ({
 };
 ```
 
+## Custom Forms using `reduxForm()` must be replaced by using `<Form>`
```
Comment (fzaninotto) [inline UPGRADE.md]:
```suggestion
## Custom Forms Using `reduxForm()` Must Be Replaced By The `<Form>` Component
```

File: UPGRADE.md [hunk 8]
```diff
@@ -374,6 +375,40 @@ export default ({
...
+                <TextInput source="note" />
++            </form>
+-            </Fragment>
++        )}
++    </Form>
++);
+-));
+```
+
+## Material-ui icons have changed
```
Comment (fzaninotto) [inline UPGRADE.md]:
```suggestion
## Material-ui Icons Have Changed
```

File: UPGRADE.md [hunk 9]
```diff
@@ -374,6 +375,40 @@ export default ({
...
+
+If you were using Material-ui icons for your design, just take care that the `v4.0` have removed some existing icons since the `v1.5`.
```
Comment (fzaninotto) [inline UPGRADE.md]:
```suggestion
If you were using Material-ui icons for your design, be aware that some icons present in 1.X versions were removed from version 4.0.
```

File: UPGRADE.md [hunk 10]
```diff
@@ -374,6 +375,40 @@ export default ({
...
+
+Example: 
+`LightbulbOutline` is no more available in `@Material-ui/icons`
```
Comment (fzaninotto) [inline UPGRADE.md]: Add line breaks for better readbility

**Pull request merged**